x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/adaptiveallocations/AdaptiveAllocationsScalerService.java

```diff
@@ -212,6 +217,10 @@ void close() {
     private volatile Scheduler.Cancellable cancellable;
     private final AtomicBoolean busy;
 
+    private final AtomicLong scaleToZeroAfterNoRequestsSeconds = new AtomicLong();
+
+    private final Set<String> inFlightScaleFromZeroRequests = new ConcurrentSkipListSet<>();
```

Comment:
Good catch!

I'd call it `deploymentIdsWithInFlightScaleFromZeroRequests` though